fvtest/threadtest/createTest.cpp

```diff
@@ -492,12 +492,12 @@ getOsStacksize(size_t stacksize)
 #if defined(LINUX) || defined(OSX)
 	/* Linux allocates 2MB if you ask for a stack smaller than STACK_MIN */
 	{
-		size_t pageSafeMinimumStack = 2 * sysconf(_SC_PAGESIZE);
+		long pageSafeMinimumStack = 2 * sysconf(_SC_PAGESIZE);
 
 		if (pageSafeMinimumStack < PTHREAD_STACK_MIN) {
 			pageSafeMinimumStack = PTHREAD_STACK_MIN;
 		}
-		if (stacksize < pageSafeMinimumStack) {
+		if (stacksize < (size_t)pageSafeMinimumStack) {
 			stacksize = pageSafeMinimumStack;
```

Comment:
`stacksize` is `size_t` and `pageSafeMinimumStack` is `int64_t`. `stacksize` is not big enough to store `pageSafeMinimumStack`. potential overflow. instead, the function signature should be updated:

```
static int64_t
getOsStacksize(uintptr_t stacksize)
```

`getOsStacksize` is used to set `create_attr_t.osStackSize`. So, the below data struct should also be updated:
```
typedef struct create_attr_t {
	int64_t osStacksize;
```

Reply:
I've update the signature in a similar way to what you suggested, although I've used [u]intptr_t since this is similar to the way omrthreadattr.c works, and the saved stacksize is uintptr_t already. What do you think?

Reply:
Yes, reasonable points. I've made your suggested changes. The comparison with PTHREAD_STACK_MIN then requires casting as the compiler believes it's a long int.

Reply:
`long int` is equivalent to `long`. `PTHREAD_STACK_MIN` is the amount of stack space that is required to start a thread. So, it won't be negative. Casting `PTHREAD_STACK_MIN` to `uintptr_t` should be fine. `uintptr_t` is a large enough data type for `PTHREAD_STACK_MIN`.